diff --git a/apps/api/src/controllers/task.controller.ts b/apps/api/src/controllers/task.controller.ts
index 36a5b63..82a371d 100644
--- a/apps/api/src/controllers/task.controller.ts
+++ b/apps/api/src/controllers/task.controller.ts
@@ -1,6 +1,18 @@
 import type { Context } from "hono";
+import { CreateTaskInput } from "../../../../packages/models/tasks/task.schema";
 import { getTaskService } from "../lib";
-import type { CreateTaskInput, Env, UpdateTaskInput } from "../types";
+import type { Env, UpdateTaskInput } from "../types";
+
+type CreateTaskContext<T> = Context<
+  {
+    Bindings: Env;
+  },
+  "/",
+  {
+    in: { json: T };
+    out: { json: any };
+  }
+>;
 
 export class TaskController {
   static async getAllTasks(c: Context<{ Bindings: Env }>) {
@@ -37,10 +49,10 @@ export class TaskController {
     }
   }
 
-  static async createTask(c: Context<{ Bindings: Env }>) {
+  static async createTask(c: CreateTaskContext<CreateTaskInput>) {
     try {
       const taskService = getTaskService(c);
-      const body: CreateTaskInput = await c.req.json();
+      const body: CreateTaskInput = c.req.valid("json");
 
       const task = await taskService.createTask(body);
 
diff --git a/apps/api/src/index.ts b/apps/api/src/index.ts
index 3ec00e3..a27a637 100644
--- a/apps/api/src/index.ts
+++ b/apps/api/src/index.ts
@@ -2,22 +2,18 @@ import { Hono } from "hono";
 import { cors } from "hono/cors";
 import { logger } from "hono/logger";
 import routes from "./routes";
-import type { Env } from "./types";
 
-const app = new Hono<{ Bindings: Env }>();
-
-app.use("*", logger());
-app.use("*", cors());
-
-app.get("/", (c) => {
-  return c.json({
-    status: "ok",
-    message: "MediPraxis API",
-    version: "1.0.0",
-  });
-});
-
-app.route("/api", routes);
+const app = new Hono()
+  .use("*", logger())
+  .use("*", cors())
+  .get("/", (c) => {
+    return c.json({
+      status: "ok",
+      message: "MediPraxis API",
+      version: "1.0.0",
+    });
+  })
+  .route("/api", routes);
 
 export type AppType = typeof app;
 
diff --git a/apps/api/src/routes/index.ts b/apps/api/src/routes/index.ts
index 7b187f3..320b277 100644
--- a/apps/api/src/routes/index.ts
+++ b/apps/api/src/routes/index.ts
@@ -2,8 +2,6 @@ import { Hono } from "hono";
 import type { Env } from "../types";
 import tasks from "./tasks";
 
-const routes = new Hono<{ Bindings: Env }>();
-
-routes.route("/tasks", tasks);
+const routes = new Hono<{ Bindings: Env }>().route("/tasks", tasks);
 
 export default routes;
diff --git a/apps/api/src/routes/tasks.ts b/apps/api/src/routes/tasks.ts
index 0bb6d93..c392bb1 100644
--- a/apps/api/src/routes/tasks.ts
+++ b/apps/api/src/routes/tasks.ts
@@ -1,14 +1,20 @@
 import { zValidator } from "@hono/zod-validator";
 import { Hono } from "hono";
+import {
+  createTaskSchema,
+  taskDetailsSchema,
+  updateTaskSchema,
+} from "../../../../packages/models/tasks/task.schema";
 import { TaskController } from "../controllers";
-import { createTaskSchema , taskDetailsSchema, updateTaskSchema } from "../../../../packages/models/tasks/task.schema";
-import type { Env } from "../types";
 
-const tasks = new Hono<{ Bindings: Env }>();
-
-tasks.get("/", TaskController.getAllTasks);
-tasks.get("/:id", zValidator("json", taskDetailsSchema), TaskController.getTaskById);
-tasks.post("/", zValidator("json", createTaskSchema), TaskController.createTask);
-tasks.put("/:id", zValidator("json", updateTaskSchema), TaskController.updateTask);
+const tasks = new Hono()
+  .post("/", zValidator("json", createTaskSchema), TaskController.createTask)
+  .get("/", TaskController.getAllTasks)
+  .get(
+    "/:id",
+    zValidator("json", taskDetailsSchema),
+    TaskController.getTaskById
+  )
+  .put("/:id", zValidator("json", updateTaskSchema), TaskController.updateTask);
 
 export default tasks;
diff --git a/apps/web-app/src/RPCTest.tsx b/apps/web-app/src/RPCTest.tsx
index 6016790..b12d8ec 100644
--- a/apps/web-app/src/RPCTest.tsx
+++ b/apps/web-app/src/RPCTest.tsx
@@ -2,11 +2,11 @@ import { createApiClient } from "../../../packages/api-client/src";
 
 const RPCTest = () => {
   // Initialize the API client - Sample code
-  const api = createApiClient(import.meta.env.VITE_API_URL);
+  const api = createApiClient(import.meta.env.VITE_API_URL).api;
 
   const testApiCall = async () => {
     try {
-      const res = await (api as any).api.tasks.$get();
+      const res = await api.tasks.$get();
       const data = await res.json();
       console.log("API Response:", data);
     } catch (error) {
@@ -16,9 +16,9 @@ const RPCTest = () => {
 
   const testCreateTaskApiCall = async () => {
     try {
-      const res = await (api as any).api.tasks.$post({
+      const res = await api.tasks.$post({
         json: {
-          task_title: "Make the log notes on the diabetic patient",
+          task_title: "Daily Blood Sugar Monitoring",
           task_type_id: "24f21ec7-bf59-4c35-9c54-36cb24afafbe",
           client_id: "24f21ec7-bf59-4c35-9c54-36cb24afafb3",
           user_id: "2a3c19b8-d352-4b30-a2ac-1cdf993d3102",
